spec/components/govuk_component/header_component_spec.rb

```diff
@@ -1,6 +1,12 @@
 require 'spec_helper'
 
 RSpec.describe(GovukComponent::HeaderComponent, type: :component) do
+  before do
+    allow_any_instance_of(GovukComponent::HeaderComponent::NavigationItem).to(
+      receive(:request).and_return(double(ActionDispatch::Request, get?: true, path: "/a-path"))
+    )
+  end
```

Comment:
Can we use something like this from the [docs](https://viewcomponent.org/guide/testing.html#setting-requestpath_parameters)?

```ruby
context "when the link is the current page" do
  it "renders an active navigation item" do
    with_request_url "/a-path" do
      component = Component.new ...

      render_inline(component)

      expect(rendered_component).to have_css "active navigation item"
    end
  end
end
```

Reply:
Missed that! That's way nicer - feel free to replace my commit.

Reply:
We tried this and it led us down to a horrible route :(

We need to add more dependencies to the gem because `with_request_url` helper requires more helpers that we do not include in the gem.

Reply:
I'm happy with this approach. I had a quick go on Friday and couldn't get it working nicely. The stub approach isn't _too bad_, it's quite clear what's going on.